com.vrcfury.vrcfury/Editor/VF/Feature/CombineMeshesBuilder.cs

```diff
@@ -0,0 +1,543 @@
+                      })
+                      .ToList();
+
+            bool IsObjectToggled(GameObject obj)
+            {
+                var path = clipBuilder.GetPath(obj.transform);
+
+                foreach (var (binding, curve) in animatedBindings)
+                {
+                    if (binding.path == path)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            bool IsObjectDisabledPermanently(GameObject obj)
+            {
+                var current = obj;
+                while (current != null && current != avatarObject)
+                {
+                    if (current.activeSelf || IsObjectToggled(obj))
+                    {
+                        return false;
+                    }
+                    current = current.transform.parent?.gameObject;
+                }
+                return true;
+            }
+
+            foreach (var (renderer, mesh, setMesh) in RendererIterator.GetRenderersWithMeshes(avatarObject))
+            {
+                if (!(renderer is SkinnedMeshRenderer skin)) continue;
+
+                if (IsObjectToggled(skin.gameObject))
+                {
+                    task.Toggled.Add(skin);
+                    continue;
+                }
+                if (IsObjectDisabledPermanently(skin.gameObject))
+                {
+                    task.Removed.Add(skin);
+                    continue;
+                }
+                var isFaceMesh = skin == faceMesh;
+                if (model.keepFaceMeshSeparate && isFaceMesh)
+                {
+                    task.NotCombinable.Add(skin);
+                    continue;
+                }
+
+                var group = new SkinnedMeshRendererGroup(skin);
+                if (!task.Groups.ContainsKey(group))
+                {
+                    task.Groups.Add(group, new List<SkinnedMeshRenderer>());
+                }
+                var entries = task.Groups[group];
+
+                var isPreferredBase = Array.IndexOf(preferredBaseMeshNames, skin.name.ToLower()) != -1;
+                var hasFaceMesh = !isFaceMesh && entries.Count > 0 && entries[0] == faceMesh;
+                if (isFaceMesh || (isPreferredBase && !hasFaceMesh))
+                {
+                    entries.Insert(0, skin);
+                }
+                else
+                {
+                    entries.Add(skin);
+                }
+            }
+
+            foreach (var entry in task.Groups.ToList())
+            {
+                if (entry.Value.Count < 2)
+                {
+                    task.Groups.Remove(entry.Key);
+                    task.NotCombinable.AddRange(entry.Value);
+                }
+            }
+
+            return task;
+        }
+
+        private ICollection<(EditorCurveBinding, AnimationCurve)> GetBindings(GameObject obj, AnimatorController controller)
```

Comment:
Is this copied from another builder? Can it be shared?

Reply:
Yep it's coming from the `BlenshapeOptimizerBuilder`, I'll move it somewhere